qiskit/algorithms/minimum_eigen_solvers/adaptvqe.py

```diff
@@ -0,0 +1,332 @@
+            # log gradients
+            if logger.isEnabledFor(logging.INFO):
+                gradlog = f"\nGradients in iteration #{str(iteration)}"
+                gradlog += "\nID: Excitation Operator: Gradient  <(*) maximum>"
+                for i, grad in enumerate(cur_grads):
+                    gradlog += f"\n{str(i)}: {str(grad[1])}: {str(grad[0])}"
+                    if grad[1] == max_grad[1]:
+                        gradlog += "\t(*)"
+                logger.info(gradlog)
+            if np.abs(max_grad[0]) < self._threshold:
+                if iteration == 1:
+                    raise QiskitError(
+                        "Gradient choice is not suited as it leads to all zero gradients gradients. "
+                        "Try a different gradient method."
+                    )
+                logger.info(
+                    "Adaptive VQE terminated successfully with a final maximum gradient: %s",
+                    str(np.abs(max_grad[0])),
+                )
+                threshold_satisfied = True
+                finishing_criterion = "Threshold converged"
+                break
+            # check indices of picked gradients for cycles
+            if self._check_cyclicity(prev_op_indices):
+                logger.info("Alternating sequence found. Finishing.")
+                logger.info("Final maximum gradient: %s", str(np.abs(max_grad[0])))
+                alternating_sequence = True
+                finishing_criterion = "Aborted due to cyclicity"
+                break
+            # add new excitation to self._ansatz
+            self._excitation_list.append(max_grad[1])
+            theta.append(0.0)
+            # run VQE on current Ansatz
+            self._tmp_ansatz.operators = self._excitation_list
+            self.ansatz = self._tmp_ansatz
+            self.initial_point = theta
+            raw_vqe_result = self.compute_minimum_eigenvalue(self._operator)
+            theta = raw_vqe_result.optimal_point.tolist()
+        else:
+            # reached maximum number of iterations
+            max_iterations_exceeded = True
+            finishing_criterion = "Maximum number of iterations reached"
+            logger.info("Maximum number of iterations reached. Finishing.")
+            logger.info("Final maximum gradient: %s", str(np.abs(max_grad[0])))
+
+        # once finished evaluate auxiliary operators if any
+        if aux_operators is not None:
+            aux_values = self._eval_aux_ops(
+                raw_vqe_result.eigenstate, aux_operators, expectation=expectation
+            )
+        else:
+            aux_values = None
+        raw_vqe_result.aux_operator_eigenvalues = aux_values
+
+        if finishing_criterion == False:
+            raise QiskitError("The algorithm finished due to an unforeseen reason!")
+
+        electronic_result = self._get_eigenstate(theta)
+
+        result = AdaptVQEResult()
+        result.combine(electronic_result)
```

Comment:
I think you would rather want to do the following:

```suggestion
        result = AdaptVQEResult()
        result.combine(raw_vqe_result)
```